lib/util/fix-owner.js

```diff
@@ -33,40 +33,38 @@ const getSelf = () => {
 
 module.exports.chownr = fixOwner
 
-function fixOwner (cache, filepath) {
+async function fixOwner (cache, filepath) {
   if (!process.getuid) {
     // This platform doesn't need ownership fixing
-    return Promise.resolve()
+    return
   }
 
   getSelf()
   if (self.uid !== 0) {
     // almost certainly can't chown anyway
-    return Promise.resolve()
+    return
   }
 
-  return Promise.resolve(inferOwner(cache)).then((owner) => {
-    const { uid, gid } = owner
+  const { uid, gid } = await inferOwner(cache)
```

Comment:
the `infer-owner` dependency here can also leverage `@npmcli/fs` in a future refactor pass